src/main/java/net/dv8tion/jda/api/utils/AttachmentProxy.java

```diff
@@ -0,0 +1,222 @@
+        Checks.notNull(file, "File");
+
+        final CompletableFuture<Path> downloadToPathFuture = downloadToPath(getUrl(width, height), file.toPath());
+        return FutureUtil.thenApplyCancellable(downloadToPathFuture, Path::toFile);
+    }
+
+    /**
+     * Downloads the data of this attachment, at the specified size, and stores it in the specified file.
+     * <br>The attachment, if an image, may be resized at any size, however if the size does not fit the ratio of the image, then it will be cropped as to fit the target size.
+     * <br>If the attachment is not an image then the size parameters are ignored and the file is downloaded.
+     *
+     * <p><b>Implementation note:</b>
+     *       The file is first downloaded into a temporary file, the file is then moved to its real destination when the download is complete.
+     *       <br>The given path can also target filesystems such as a ZIP filesystem.
+     *
+     * @param  path
+     *         The file in which to download the image
+     * @param  width
+     *         The width of this image, must be positive
+     * @param  height
+     *         The height of this image, must be positive
+     *
+     * @return a {@link CompletableFuture} which holds a {@link Path}, it is the same as the path passed in the parameters.
+     *
+     * @throws IllegalArgumentException
+     *         If any of the follow checks are true
+     *         <ul>
+     *             <li>The target path is null</li>
+     *             <li>The parent folder of the target path does not exist</li>
+     *             <li>The requested width is negative or 0</li>
+     *             <li>The requested height is negative or 0</li>
+     *         </ul>
+     */
+    @Nonnull
+    public CompletableFuture<Path> downloadToPath(@Nonnull Path path, int width, int height)
+    {
+        Checks.notNull(path, "Path");
+
+        return downloadToPath(getUrl(width, height), path);
+    }
+
+    /**
+     * Downloads the data of this attachment, at the specified size, and constructs an {@link Icon} from the data.
+     * <br>The attachment, if an image, may be resized at any size, however if the size does not fit the ratio of the image, then it will be cropped as to fit the target size.
+     * <br>If the attachment is not an image then the size parameters are ignored and the file is downloaded.
+     *
+     * @param  width
+     *         The width of this image, must be positive
+     * @param  height
+     *         The height of this image, must be positive
+     *
+     * @return a {@link CompletableFuture} which holds an {@link Icon}.
+     *
+     * @throws IllegalArgumentException
+     *         If any of the follow checks are true
+     *         <ul>
+     *             <li>The requested width is negative or 0</li>
+     *             <li>The requested height is negative or 0</li>
+     *         </ul>
+     */
+    @Nonnull
+    public CompletableFuture<Icon> downloadAsIcon(int width, int height) {
```

Comment:
```suggestion
    public CompletableFuture<Icon> downloadAsIcon(int width, int height)
    {
```

Reply:
There should also be an overload without width and height parameters, since that is possible in the api as well.